### DISCUSSION TITLE
Chrome Advanced Web Apps Fund sponsorship

### DISCUSSION THREAD

**Discussion by ganigeorgiev** [pocketbase/pocketbase#730] [Announcements]
I'm happy to announce that PocketBase was approved to receive a grant of $20,000 as part of the [Chrome Advanced Web Apps Fund](https://developer.chrome.com/blog/advanced-web-apps-fund/).

**The TLDR version is that starting from November I'll be able to work full-time on PocketBase for at least 5 months.**

## Short-term and Long-term goals

**The current short term goal is to have a v0.8.0-alpha pre-release within the next 2-3 weeks.**

The v0.8.x release will come with a lot of changes (_some of which unfortunately are breaking_), but here is a list of what you could expect in terms of features:

- simplified users management #376
- support for different auth model collections allowing full "manager-subordinate" accounts management
- username/password auth support #120
- indirect relation expansions #312
- `json` field props filter and sort support https://github.com/pocketbase/pocketbase/issues/423#issuecomment-1258302125
- simplified API preview docs and including the auth actions for the new "auth" collection type
- other various Admin UI tweaks and minor performance improvements

In short, the plan for now is:

- finalize the remaining Go integration and JavaScript/Dart SDKs unit tests
- prepare a v0.7.x -> v0.8.x migration guide
- **publish v0.8.0-alpha pre-release**
- while collecting community feedback for the new api during the test/pre-release period (I'll wait for ~2-3 weeks), start the work on the new documentation and simplify its editing process to allow external contributions
- check if Apple sign-in could be "squeezed" in v0.8.0 since it currently represents a blocker for iOS mobile developers https://github.com/pocketbase/pocketbase/issues/202#issuecomment-1242652375
- **release v0.8.0**

**The long term goal is to complete the [v1.0.0 roadmap](https://github.com/orgs/pocketbase/projects/2) and to spend more time researching 3 specific topics:**

1. How to simplify the collections migration between environments [[#489](https://github.com/pocketbase/pocketbase/discussions/489)]

2. Export/import of records data from/to different sources (csv, json, etc.) https://github.com/pocketbase/pocketbase/issues/48#issuecomment-1211775627

3. **Workflow builder in the Admin UI** (this will probably replace and extend the existing [Collection triggers](https://github.com/pocketbase/pocketbase/issues/317) feature request; I'll try to add some implemention details after v0.8 release)

## What does this mean for Presentator

I didn't forget about [Presentator](https://github.com/presentator/presentator) (after all PocketBase was initially created because of it).
Once we tackle the above issues and stabilize the PocketBase API, I'll split my time between the 2 open source projects and will try to meet the initial development schedule from https://github.com/presentator/presentator/issues/183. I'll make sure to document the process so that Presentator could be used also as an example for mid-size PocketBase app.

---

That's it :). Special shout-out to the [Google/Chrome Advanced Apps Fund team](https://opencollective.com/chrome/projects/advanced-apps-fund) and to everyone else who contributed so far (financially, by reporting bugs or making PRs). Thank you!

**Comment by benallfree** [comment]
I love this!! Congratulations @ganigeorgiev! If I get to #317 and #468 (or pieces of them) before you do, I'll send PR's :)

**Comment by enstyled** [comment]
Amazing, congratulations!

**Comment by jm61** [comment]
Thankfully, there is still fairness in this world!

**Comment by julianruizsa** [comment]
Congrats! Pocketbase is an awesome project.

**Comment by pacholoamit** [comment]
Awseome! Pocketbase is truly an amazing piece of software

**Comment by corvalan-roberto** [comment]
Es una inmensa alegría, saber que vamos a tener Pocketbase para mucho tiempo mas, El aporte que le haces a la comunidad es grande. !Felicitaciones¡

**Comment by paulocoutinhox** [comment]
Amazing. Will be nice the reorder fields in the Admin UI. Congratulations!

**Comment by ollema** [comment]
amazing news, congratulations!

**Comment by jimafisk** [comment]
This is so wonderful! Congrats @ganigeorgiev!

**Comment by brewhousedigital** [comment]
Congrats! So exciting!

**Comment by angeldev96** [comment]
I'm really happy about this, pocketbase has a lot of potential.

**Comment by DigNZ** [comment]
Just found this. Congratulations. I love pocketbase so much my colleagues are getting sick of me going on about it. Thanks for all your hardwork and great to see it rewarded and an exciting roadmap in place.

**Comment by benallfree** [comment]
@ganigeorgiev It has been 5 months, can you share future plans? Do you need more financial support to keep going?

**Reply by ganigeorgiev** [reply]
The general plan hasn't changed and I'm still working on the [tasks for v1](https://github.com/orgs/pocketbase/projects/2) (_alongside the new design and frontned for Presentator v3_).

> Do you need more financial support to keep going?

No, financial support definitely helps but it is not something that is required for me to continue working on the project (_I didn't expected to receive the grant in the first place_). I work on PocketBase because I need it. The grant just helps speeding the process.

I'm not planning starting a new job at least before releasing Presentator v3 and its new Figma and XD plugins, which I hope to finalize everything somewhere before August (_technically we already have what we need for the backend part, I'm just terrible at multitasking; I'm trying to spend now a little more time on polishing it since later once I start a new job I imagine all my free time I'll be busy dealing mostly with PocketBase issues_).

The short term PocketBase priorities for now are:

- (v0.16.0) finalize the ongoing auto backup and restore apis and tests
- (v0.16.0) add selfupdate cli command for easier application upgrades when using the prebuilt executable
- prepare a testbed and benchmarks so that we have something to compare to before the planned hooks refactoring and goja app hooks integration
- (v0.17.0) refactor the hooks and some of the related internals (_once I have a more clear idea of the changes I'll post an announcements for feedback_)
- (v0.17.0) add support for goja hooks and dynamic code loading
- (v0.17.0) better and more complete "Use as framework" docs (_we'll need docs for both the JS and Go; I still haven't decided how that will be managed and whether it would be worth investing the time to automate some part of it_)

**Comment by vipas84** [comment]
please continue! its amazing!
